pkg/spoke/controllers/appliedmanifestcontroller/appliedmanifestwork_controller.go

```diff
@@ -201,19 +201,26 @@ func (m *AppliedManifestWorkController) syncManifestWork(
 }
 
 // findUntrackedResources returns applied resources which are no longer tracked by manifestwork
+// API version should be ignored when checking if a resource is no longer tracked by a manifestwork.
+// This is because we treat resources of same GroupResource but different version equivalent.
+// It also compares UID of the appliedResources to identify the untracked appliedResources because
+// 1. The UID should keep the same for resources with different versions.
+// 2. The UID in the newAppliedResources is always the latest updated one. The only possibility that UID
+// in appliedResources differs from what in newAppliedResources is that this resource is recreated.
+// Its UID in appliedResources is invalid hence recording it as untracked applied resource and delete it is safe.
```

Comment:
Doesn't this change the semantics of co-ownership of a resource and resource adoption?  It seems like this would result in controllers that keep trying to delete the shared resource.

Reply:
If the resource is shared by manifestwork/foo and manifestwork/bar. Deleting manifestwork/foo will cause resource deleted and recreated once.  It is because we do not update resource uid in deleting appliedmanifestwork. So deleting appliedmanifeswork will not make it to delete the newly created shared resource.

We may consider do not delete the shared resource at all in another enhancement? For instance, a shared resource could have multiple ownerrefs of appliemanifestworks, and if one of them is deleted. We remove the ownerref instead of deleting the resource.